tests/functional/cron/testReindex.js

```diff
@@ -213,60 +215,82 @@ describe('UtapiReindex', () => {
             });
         }
 
-        beforeEach(done => {
-            reindex._scheduleJob();
-            // Wait until the scripts have started and finished reindexing.
-            async.series([
-                next => waitUntilLockHasValue({ value: 'true' }, next),
-                next => waitUntilLockHasValue({ value: null }, next),
-            ], done);
-        });
-
-        it('should reindex metrics', done => {
-            async.parallel([
-                next => {
-                    const params = {
-                        resource: {
-                            type: 'buckets',
-                            buckets: [mock.values.BUCKET_NAME],
+        const bucketCounts = [1, 1001];
+
+        bucketCounts.forEach(count => {
+            describe(`bucket listing with a length of ${count}`, () => {
+                const bucket = `${mock.values.BUCKET_NAME}-${count - 1}`;
+                const MPUBucket = `${constants.mpuBucketPrefix}${bucket}`;
+
+                beforeEach(done => {
+                    bucketD.setBucketContent({
+                        bucketName: bucket,
+                        contentLength: 1024,
+                    })
+                    .setBucketContent({
+                        bucketName: MPUBucket,
+                        contentLength: 1024,
+                    })
+                    .setBucketCount(count)
+                    .createBuckets();
+
+                    reindex._scheduleJob();
+
+                    // Wait until the scripts have finished reindexing.
+                    async.series([
+                        next => waitUntilLockHasValue({ value: 'true' }, next),
```

Comment:
Might be racy here if the reindex script finishes before the first check for `value: 'true'` that could lead to the test being stuck here (although maybe unlikely in practice)

Reply:
Good point. There is definitely potential for that here however unlikely because the single bucket listing test takes hundreds of MS. Good to prevent it from ever occurring though, so I'll update such that at least one call to Redis is made before the job is run.